### PR TITLE
feat: handle temporary guest users details (WPB-10454)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10454" title="WPB-10454" target="_blank">TaskWPB-10454</a>  [Android] Handle temporary guest users
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We don't have handling for temporary users

### Causes (Optional)

Not implemented

### Solutions

Implement the handling for temporary users; avatar, handle, etc.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Join a conversation as a guest user, you should see indicators about; time left, avatar with indicator.

### Notes

Unrelated changes:
- merging tests from develop about username mapping
- marking the adr leftovers, action and document

### Attachments (Optional)

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
# New ADR(s) in this PR 📚:
# 3. Use parameterizable tests in JUnit5

Date: 2024-08-05

## Status

Accepted

## Context

Sometimes we need to write multiple tests for the same scenario, changing only the input values.

## Decision

We will use parameterizable tests in JUnit5 to avoid writing multiple tests for the same scenario.

## Consequences

- Introduction of `@ParameterizedTest` annotation in the test class
  and [library](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests).
- The test method will receive the parameters as arguments.

File: app/src/main/kotlin/com/wire/android/ui/userprofile/common/UserProfileInfo.kt [hunk 1]
```diff
@@ -231,6 +240,16 @@ fun UserProfileInfo(
     }
 }
 
+@SuppressLint("ComposeNamingLowercase")
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/common/UserProfileInfo.kt]: Is this being used? 🤔 

As the composable function starts with an Uppercase letter, or it wants something like: `UserName`? 
just curious

**Review by alexandreferris — APPROVED** [review]
Amazing work!

I just have one question but not a blocker, mostly curious hehe

File: app/src/main/kotlin/com/wire/android/ui/userprofile/common/UserProfileInfo.kt (hunk 1 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/ui/userprofile/common/UserProfileInfo.kt]: Good catch, probably a leftover

**Review by Garzas — APPROVED** [review]
👏

**Pull request merged**